.github/workflows/release.yml

```diff
@@ -88,11 +88,17 @@ jobs:
 
       - name: Update PR with changelog
         run: |
-          git add CHANGELOG.md
           git config --global user.name "PasteurBot"
           git config --global user.email "hello@simulation.science"
-          git commit -m "chore: update changelog before release"
-          git push origin HEAD:${{ github.head_ref }}
+
+          git add CHANGELOG.md
+
+          if git diff --cached --quiet; then
+            echo "No changes to commit"
+          else
+            git commit -m "chore: update changelog before release"
+            git push origin HEAD:${{ github.head_ref }}
```

Comment:
Seems like this line didn't work in the PR either, since `github.head_ref` is empty?

Reply:
it seems so; maybe it should just be github.ref?

Reply:
The issue is that this is not reacting to a pull request event, so we can't know which branch the commit is on without explicitly determining that.

Reply:
I don't know.

In general, commits can belong to multiple branches so I'm not sure how this is handled. According to the docs, `ref` should work though.